### PR TITLE
Refactor books application to use Classes

### PR DESCRIPTION
**Pull request by Jord4N-AC**
This project was made pair-programming, In this project, I and my coding partners refactored the Awesome books application to use JavaScript classes, We also make sure to meet the following requirements:

### Project Requirements
- Use JavaScript classes instead of objects and arrays (or use them  inside classes)
- Create class methods to add and remove books
- Maintain all of the original functionality of the application (i.e. no bugs were introduced while refactoring)
- Don't use any library or framework, use only plain JavaScript
- Add CSS styles to the application to make it match this wireframe

### General Requirements
- Check linter errors
- Use the correct GitHub flow
- Document my work in a professional way
- Follow best practices for HTML, CSS & JavaScript
- Open a Pull Request and ask for a review

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,56 +1,78 @@
...
+    const newBook = new BookObject(inputTitle.value, inputAuthor.value, theBooks.length);
+    newBook.add();
+    theBooks.push(newBook);
+
     localStorage.setItem('booksArray', JSON.stringify(theBooks));
 
-    inputTitle.value = '';
-    inputAuthor.value = '';
+    // inputTitle.value = '';
+    // inputAuthor.value = '';
```
Comment (TedLivist) [inline index.js]: - [ ] Kindly remove commented code. It makes your codebase look messy and hard to read.

File: index.js [hunk 2]
```diff
@@ -60,10 +82,10 @@ function addBooks() {
...
 function loadBooks() {
   JSON.parse(localStorage.getItem('booksArray')).forEach((book, i) => {
-    creaateAndAppend(book.title, book.author, i);
+    const newBook = new BookObject(book.title, book.author, i);
+    newBook.add();
+    theBooks.push(newBook);
   });
-
-  theBooks = JSON.parse(localStorage.getItem('booksArray'));
 }
```
Comment (TedLivist) [inline index.js]: - [ ] Your methods for implementing the functionalities should be within the **BookObject** class. Kindly move this 
**`loadBooks()`** method into the class.

File: index.js [hunk 3]
```diff
@@ -1,56 +1,78 @@
...
+        book.id = i;
+      });
+      localStorage.setItem('booksArray', JSON.stringify(theBooks));
+    }
+
+    element.addEventListener('click', removeBook);
+  }
 }
 
 function addBooks() {
```
Comment (TedLivist) [inline index.js]: - [ ] Kindly move this **`addBooks()`** method into the **BookObject** class.

**Review by TedLivist — COMMENTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Jord4N-AC, @Brenda309 and @AbrahaKahsay,

Good job so far!
✔️ Good documentation
✔️ Nice design
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 2 above)
Comment (Jord4N-AC) [reply index.js]: I have the method inside the class, 
`loadBooks()` is a function to load the books from the data in `localStorage`
`const newBook = new BookObject(book.title, book.author, i);`
`newBook.add();`
Are usin the class and method class
I don't know if you check or understand the code??
I don't understand your feedback

File: index.js (hunk 3 above)
Comment (Jord4N-AC) [reply index.js]: `addBooks()` equals to function that is used to add books when pressing Add button
I'm using the class and class method inside `addBooks()`

File: index.js (hunk 1 above)
Comment (Jord4N-AC) [reply index.js]: Most of the code reviewers mark this as optional

**Comment by Jord4N-AC** [comment]
If I put everything in the class How I'm going to use or call that to use in an `eventListener`, or to use when the page loads??
I don't really understand

**Comment by TedLivist** [comment]
@Jord4N-AC, you can move them into the class as `static` methods and then call the methods on the class.

```
class BookObject
  ...
  static addBooks() {
    ...
```

For the event listener, call the method on the class like so:
```
addButton.addEventListener('click', BookObject.addBooks);
```
You can do the same for loadBooks.

**Comment by Jord4N-AC** [comment]
@TedLivist thanks  a lot

**Comment by Amrhub** [comment]
Thank you @TedLivist and @Jord4N-AC  for the awesome communication ❤️

**Review by Amrhub — APPROVED** [review]
# Approved 💯 ✅ 🥳 🎈 🎉 

Hi @Jord4N-AC, 👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights ✅ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**